v03_pipeline/lib/reference_datasets/clinvar.py

```diff
@@ -0,0 +1,189 @@
+    with tempfile.NamedTemporaryFile(
+        suffix='.txt.gz',
+        delete=False,
+    ) as tmp_file, requests.get(
+        CLINVAR_SUBMISSION_SUMMARY_URL,
+        stream=True,
+        timeout=10,
+    ) as r:
+        shutil.copyfileobj(r.raw, tmp_file)
+    ht = hl.import_table(
+        tmp_file.name,
+        force=True,
+        filter='^(#[^:]*:|^##).*$',  # removes all comments except for the header line
+        types={
+            '#VariationID': hl.tstr,
+            'Submitter': hl.tstr,
+            'ReportedPhenotypeInfo': hl.tstr,
+        },
+        missing='-',
+    )
+    ht = ht.rename({'#VariationID': 'VariationID'})
+    ht = ht.select('VariationID', 'Submitter', 'ReportedPhenotypeInfo')
+    return ht.group_by('VariationID').aggregate(
+        Submitters=hl.agg.collect(ht.Submitter),
+        Conditions=hl.agg.collect(ht.ReportedPhenotypeInfo),
+    )
+
+
+def select_fields(ht):
+    clnsigs = parsed_clnsig(ht)
+    ht = ht.select(
+        alleleId=ht.info.ALLELEID,
+        pathogenicity=hl.if_else(
+            CLINVAR_PATHOGENICITIES_LOOKUP.contains(clnsigs[0]),
+            clnsigs[0],
+            CLINVAR_DEFAULT_PATHOGENICITY,
+        ),
+        assertions=hl.if_else(
+            CLINVAR_PATHOGENICITIES_LOOKUP.contains(clnsigs[0]),
+            clnsigs[1:],
+            clnsigs,
+        ),
+        # NB: there's a hidden enum-mapping inside this clinvar function.
+        conflictingPathogenicities=parsed_and_mapped_clnsigconf(ht),
+        goldStars=CLINVAR_GOLD_STARS_LOOKUP.get(hl.delimit(ht.info.CLNREVSTAT)),
+        submitters=ht.submitters,
+        # assumes the format 'MedGen#:condition', e.g.'C0023264:Leigh syndrome'
+        conditions=hl.map(
+            lambda p: p.split(r':')[1],
+            ht.conditions,
+        ),
+    )
+    return ht.transmute(
+        pathogenicity_id=enum_map(ht.pathogenicity, CLINVAR_PATHOGENICITIES),
+        assertion_ids=enum_map(ht.assertions, CLINVAR_ASSERTIONS),
+    )
+
+
+def get_ht(
+    clinvar_url: str,
+    reference_genome: ReferenceGenome,
+    dataset_type: DatasetType,
+) -> hl.Table:
+    version = parse_clinvar_release_date(clinvar_url)
+    with tempfile.NamedTemporaryFile(
+        suffix='.vcf.gz',
+        delete=False,
+    ) as tmp_file, requests.get(clinvar_url, stream=True, timeout=10) as r:
+        shutil.copyfileobj(r.raw, tmp_file)
+    ht = hl.import_vcf(
+        tmp_file.name,
+        reference_genome=reference_genome.value,
+        drop_samples=True,
+        skip_invalid_loci=True,
+        contig_recoding=reference_genome.contig_recoding(include_mt=True),
+        force_bgz=True,
+    ).rows()
+    ht = filter_contigs(ht, reference_genome, dataset_type)
+    submitters_ht = get_submission_summary_ht()
+    ht = ht.annotate(
+        submitters=submitters_ht[ht.rsid].Submitters,
+        conditions=submitters_ht[ht.rsid].Conditions,
+    )
+    ht = select_fields(ht)
+    return ht.annotate_globals(
```

Comment:
theres a  very similar call to `annotate_globals` in the `create_table` method of the update task. I feel like because we have such a consistent schema for how we want globals to be annotated it would be better to have this logic shared/ generalized in some way

Reply:
one related question: should the update task be doing any globals annotation or should that happen only in the `get_ht` scripts?

Reply:
yeah, I also had this question and thought.  I didn't love the enum magic we were doing to try to fit `clinvar` into the config structure last time around, so felt it was better to avoid putting `enums` in the config and just `annotate` manually in the `get_ht` functions.  It's not a bad idea to _also_ put the `enums` in the config so that we can share the `globals` logic downstream.

Reply:
agreed that putting them in the config and sharing the logic is worth doing even if we don't use it in the `get_ht` function

Reply:
other than this I think all my comments have been resolved!

Reply:
yep!  It's in progress.